### PR TITLE
Update NodeJS example with Query Service

### PR DESCRIPTION
**Pull request by Zork33**

### CODE REVIEW

**Comment by blinkov** [comment]
@Zork33 this has conflicts that need to be resolved

**Comment by blinkov** [comment]
@Zork33 do you intend to finish this PR?

**Comment by Zork33** [comment]
Yes, I will fix conflicts

**Comment by Zork33** [comment]
Conflicts are fixed

File: ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md [hunk 1]
```diff
@@ -1,261 +1,353 @@
...
+  ```
+
+{% endlist %}
 
 {% include [param_queries.md](steps/06_param_queries.md) %}
 
-Here's a code sample that shows how to use the `Session.executeQuery()` method with the queries and parameters
-prepared by `Session.prepareQuery()`.
+There is no explicit Prepared Query option in the Query Service. YDB determines the necessity of using this mode by
+itself.
```
Comment (blinkov) [inline ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md:198]:
```suggestion
```
We are talking about the query service in documentation only if it is absolutely necessary.

File: ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md [hunk 2]
```diff
@@ -1,261 +1,353 @@
...
+                               air_date
+                        FROM episodes
+                        WHERE series_id = $seriesId
+                          AND season_id = $seasonId
+                          AND episode_id = $episodeId;`
+                });
+                const {value: resultSet} = await resultSets.next();
+                const {value: row} = await resultSet.rows.next();
+                await opFinished;
+                logger.info(`Select prepared query ${JSON.stringify(row, null, 2)}`);
```
Comment (blinkov) [inline ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md:228]:
```suggestion
                logger.info(`Parametrized select query ${JSON.stringify(row, null, 2)}`);
```

File: ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md [hunk 3]
```diff
@@ -1,261 +1,353 @@
...
-    const query = `
-        ${SYNTAX_V1}
-        DECLARE $value AS Utf8;
-
-        SELECT key
-        FROM ${TABLE}
-        WHERE value = $value;`;
-
-    logger.info('Making a stream execute scan query...');
+`QuerySession.execute()` method is used to retrieve data by a stream.
```
Comment (blinkov) [inline ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md:237]:
```suggestion
The `QuerySession.execute()` method is used to retrieve data in a streaming manner.
```

File: ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md [hunk 4]
```diff
@@ -1,261 +1,353 @@
...
-
-    let count = 0;
-    await session.streamExecuteScanQuery(query, (result) => {
-        logger.info(`Stream scan query partial result #${++count}: ${formatPartialResult(result)}`);
-    }, params);
-
-    logger.info(`Stream scan query completed, partial result count: ${count}`);
+```ts
+async function selectWithParametrs(driver: Driver, data: ThreeIds[], logger: Logger): Promise<void> {
+    logger.info('Selecting prepared query...');
```
Comment (blinkov) [inline ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md:241]:
```suggestion
    logger.info('Selecting with a parametrized query...');
```

File: ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md [hunk 5]
```diff
@@ -1,261 +1,353 @@
...
+                               air_date
+                        FROM episodes
+                        WHERE series_id = $seriesId
+                          AND season_id = $seasonId
+                          AND episode_id = $episodeId;`
+                });
+                const {value: resultSet} = await resultSets.next();
+                const {value: row} = await resultSet.rows.next();
+                await opFinished;
+                logger.info(`Select prepared query ${JSON.stringify(row, null, 2)}`);
```
Comment (blinkov) [inline ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md:269]:
```suggestion
                logger.info(`Parametrized select query ${JSON.stringify(row, null, 2)}`);
```

File: ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md [hunk 6]
```diff
@@ -1,261 +1,353 @@
...
-        logger.info(`Executing query with txId ${txId}.`);
-        await session.executeQuery(preparedQuery, params, {txId});
-        await session.commitTransaction({txId});
-        logger.info(`TxId ${txId} committed.`);
-    }
-    await withRetries(update);
-}
-```
+Here's a code sample that demonstrates how to explicitly use the `Session.beginTransaction()`
+and `Session.сommitTransaction()` calls to create and terminate a transaction:
```
Comment (blinkov) [inline ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md:279]:
```suggestion
Here's a code sample demonstrating how to explicitly use the `Session.beginTransaction()` and `Session.commitTransaction()` calls to execute a transaction:
```

File: ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md [hunk 7]
```diff
@@ -1,261 +1,353 @@
...
+
+{% list tabs %}
+
+- do()
+
+  ```ts
+  async function explicitTcl(driver: Driver, ids: ThreeIds, logger: Logger) {
+      logger.info('Running prepared query with explicit transaction control...');
```
Comment (blinkov) [inline ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md:287]:
```suggestion
      logger.info('Running a parametrized query with explicit transaction control...');
```

File: ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md [hunk 8]
```diff
@@ -1,261 +1,353 @@
...
+          }
+      });
+  }
+  ```
+
+- doTx()
+
+  ```ts
+  async function transactionPerWholeDo(driver: Driver, ids: ThreeIds, logger: Logger) {
+      logger.info('Running query with one transaction per whole doTx()...');
```
Comment (blinkov) [inline ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md:322]:
```suggestion
      logger.info('Running a query with one transaction per whole doTx()...');
```

**Review by blinkov — DISMISSED** [review]

File: ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md [hunk 9]
```diff
@@ -1,263 +1,352 @@
 # Приложение на Node.js
 
-На этой странице подробно разбирается код [тестового приложения](https://github.com/ydb-platform/ydb-nodejs-sdk/tree/master/examples/basic-example-v1), доступного в составе [Node.js SDK](https://github.com/ydb-platform/ydb-nodejs-sdk) {{ ydb-short-name }}.
+На этой странице представлено подробное описание кода тестового приложения
+[basic-example-v2-with-query-service](https://github.com/ydb-platform/ydb-nodejs-sdk/tree/master/examples/basic-example-v2-with-query-service),
+доступого в составе [Node.js SDK](https://github.com/ydb-platform/ydb-nodejs-sdk) {{ ydb-short-name }}.
```
Comment (blinkov) [inline ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md:5]:
```suggestion
На этой странице представлено подробное описание кода 
[тестового приложения](https://github.com/ydb-platform/ydb-nodejs-sdk/tree/master/examples/basic-example-v2-with-query-service),
доступого в составе [{{ ydb-short-name }} Node.js SDK](https://github.com/ydb-platform/ydb-nodejs-sdk) {{ ydb-short-name }}.
```

File: ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md [hunk 10]
```diff
@@ -1,263 +1,352 @@
...
-    logger.info(`selectSimple result: ${JSON.stringify(result, null, 2)}`);
-}
-```
+{% endlist %}
 
 {% include [param_queries.md](steps/06_param_queries.md) %}
 
-Фрагмент кода, приведенный ниже, демонстрирует использование подготовленных с помощью `Session.prepareQuery()` запросов и параметров
-в методе `Session.executeQuery()`.
+В Query Service нет явной опции Prepared Query.  YDB определяет необходимость использование этого режима самостоятельно.
```
Comment (blinkov) [inline ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md:198]:
```suggestion
```

File: ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md [hunk 11]
```diff
@@ -1,263 +1,352 @@
...
-        SELECT key
-        FROM ${TABLE}
-        WHERE value = $value;`;
-
-    logger.info('Making a stream execute scan query...');
-
-    const params = {
-        '$value': TypedValues.utf8('odd'),
-    };
+В Query Service для получения данных потоком используется метод `QuerySession.execute()`.
```
Comment (blinkov) [inline ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md:237]:
```suggestion
Для получения данных потоком используется метод `QuerySession.execute()`.
```

File: ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md [hunk 12]
```diff
@@ -1,263 +1,352 @@
...
+              logger.info(`Select prepared query ${JSON.stringify(row, null, 2)}`);
+          }
+      }
+  });
 }
 ```
 
 {% include [transaction-control.md](steps/10_transaction_control.md) %}
 
 Фрагмент кода, демонстрирующий явное использование вызовов `Session.beginTransaction()` и `Session.сommitTransaction()` для создания и завершения транзакции:
```
Comment (blinkov) [inline ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md:278]:
```suggestion
Фрагмент кода, демонстрирующий явное использование вызовов `Session.beginTransaction()` и `Session.сommitTransaction()` для выполнения транзакции:
```

File: ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md [hunk 13]
```diff
@@ -1,263 +1,352 @@
...
+    logger.info('Upsert completed.')
 }
 ```
 
 {% include [steps/04_query_processing.md](steps/04_query_processing.md) %}
 
-Для выполнения YQL-запросов используется метод `Session.executeQuery()`.
+Для выполнения YQL-запросов используется метод `QuerySession.execute()`.
+
+В зависимости оп параметра `rowMode` данные можно получить в JavaScript  форме или как {{ ydb-short-name }} структуры.
```
Comment (blinkov) [inline ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md:133]:
```suggestion
В зависимости оп параметра `rowMode` данные можно получить в JavaScript форме или как {{ ydb-short-name }} структуры.
```

File: ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md [hunk 14]
```diff
@@ -1,263 +1,352 @@
...
+                             air_date
+                      FROM episodes
+                      WHERE series_id = $seriesId
+                        AND season_id = $seasonId
+                        AND episode_id = $episodeId;`
+              });
+              const {value: resultSet} = await resultSets.next();
+              const {value: row} = await resultSet.rows.next();
+              await opFinished;
+              logger.info(`Select prepared query ${JSON.stringify(row, null, 2)}`);
```
Comment (blinkov) [inline ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md:228]:
```suggestion
              logger.info(`Parametrized select query ${JSON.stringify(row, null, 2)}`);
```

File: ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md [hunk 15]
```diff
@@ -1,263 +1,352 @@
...
```
Comment (blinkov) [inline ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md:269]:
```suggestion
              logger.info(`Parametrized select query ${JSON.stringify(row, null, 2)}`);
```

File: ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md [hunk 16]
```diff
@@ -1,263 +1,351 @@
 # Приложение на Node.js
 
-На этой странице подробно разбирается код [тестового приложения](https://github.com/ydb-platform/ydb-nodejs-sdk/tree/master/examples/basic-example-v1), доступного в составе [Node.js SDK](https://github.com/ydb-platform/ydb-nodejs-sdk) {{ ydb-short-name }}.
+На этой странице представлено подробное описание кода 
+[тестового приложения](https://github.com/ydb-platform/ydb-nodejs-sdk/tree/master/examples/basic-example-v2-with-query-service),
+доступого в составе [{{ ydb-short-name }} Node.js SDK](https://github.com/ydb-platform/ydb-nodejs-sdk) {{ ydb-short-name }}.
```
Comment (blinkov) [inline ydb/docs/ru/core/dev/example-app/_includes/example-nodejs.md:5]:
```suggestion
На этой странице представлено подробное описание кода
[тестового приложения](https://github.com/ydb-platform/ydb-nodejs-sdk/tree/master/examples/basic-example-v2-with-query-service),
доступого в составе [{{ ydb-short-name }} Node.js SDK](https://github.com/ydb-platform/ydb-nodejs-sdk) {{ ydb-short-name }}.
```

**Review by fomichev3000 — APPROVED** [review]

**Pull request merged**